packages/terra-spacer/src/_spacerShorthandUtils.js

```diff
@@ -0,0 +1,99 @@
+import { SpacerSizes } from './_spacerConstants';
+
+const SHORTHAND_DELIMITER = ' ';
+const shLengthBoundaries = { min: 1, max: 4 };
+
+const conflictingPropNamesError = (
+  propName = '<SUPPLY SHORTHAND PROP NAME>',
+  conflictingPropName = '<SUPPLY CONFLICTING PROP NAME>',
+  componentName = 'Spacer',
+) => new Error(`Expected only ${propName} or ${conflictingPropName} to be supplied to ${componentName} but both were provided. Validation failed.`);
+
+const invalidShArgsLengthError = (
+  propName = '<SUPPLY SHORTHAND PROP NAME>',
+  valuesLength = '<SUPPLY RECEIVED ARGS LENGTH>',
+  minLength = shLengthBoundaries.min,
+  maxLength = shLengthBoundaries.max,
+  componentName = 'Spacer',
+) => new Error(`Invalid number of argument substrings provided to ${propName} in ${componentName}. Expected ${minLength}-${maxLength}. Received ${valuesLength}. Validation failed.`);
+
+const invalidShArgValuesError = (
+  propName = '<SUPPLY SHORTHAND PROP NAME>',
+  invalidShValues = [],
+  expectedValues = Object.values(SpacerSizes),
+  componentName = 'Spacer',
+) => new Error(`Expected ${propName} shorthand values to be one of ${expectedValues} but "${invalidShValues}" supplied to ${componentName}. Validation Failed.`);
+
+export const ShorthandPropErrors = {
+  conflictingPropName: conflictingPropNamesError,
+  invalidShorthandLength: invalidShArgsLengthError,
+  invalidShorthandArg: invalidShArgValuesError,
+};
+
+export const mapShorthandToObject = (propName = '', propVal = '') => {
+  const shValues = propVal.split(SHORTHAND_DELIMITER);
+
+  const generateShObject = (top, right, bottom, left) => ({
+    [`${propName}Top`]: top,
+    [`${propName}Right`]: right,
+    [`${propName}Bottom`]: bottom,
+    [`${propName}Left`]: left,
+  });
+
+  switch (shValues.length) {
+    case 1:
+      return generateShObject(shValues[0], shValues[0], shValues[0], shValues[0]);
+    case 2:
+      return generateShObject(shValues[0], shValues[1], shValues[0], shValues[1]);
+    case 3:
+      return generateShObject(shValues[0], shValues[1], shValues[2], shValues[1]);
+    case 4:
+      return generateShObject(shValues[0], shValues[1], shValues[2], shValues[3]);
+    default:
+      return {};
+  }
+};
+
+export const shorthandValidator = (props = {}, propName = '') => {
```

Comment:
Can we export everything in the Utils at the bottom so we know what is available without scrolling or searching on export?